File: huntthewampus.py
from sys import stdin,stdout

s = int(stdin.readline())
points = set()
while len(points) < 4:
    s += s // 13 + 15
    d = s % 100
    points.add((d // 10, d % 10))

moves = 0
while points:
    L = int(stdin.readline())
    x2,y2 = L // 10, L % 10
    if (x2,y2) in points:
        stdout.write("You hit a wumpus!\n")
        points.remove((x2,y2))
    if points:
        lowest = 100000000
        for x,y in points:
            lowest = min(lowest, abs(x - x2) + abs(y - y2))
        stdout.write(f"{lowest}\n")
    moves += 1
stdout.write(f"Your score is {moves} moves.\n")

File: huntthewampus.cpp
#include <bits/stdc++.h>
#include <unordered_set>

using namespace std;

typedef unordered_set<int> usi;

int main() {
    int s; scanf("%d",&s);
    usi points;
    // generate 4 random unique wumpus locations
    while (points.size() < 4) {
        s += s / 13 + 15;
        points.insert(s % 100);
    }
    int moves = 0; int loc;
    while (scanf("%d",&loc) == 1) {
        if (points.find(loc) != points.end()) {
            printf("You hit a wumpus!\n");
            points.erase(loc);
        }
        if (!points.empty()) {
            int lowest = 1e9;
            for (int p: points) lowest = min(lowest, abs(p / 10 - loc / 10) + abs(p % 10 - loc % 10));
            printf("%d\n",lowest);
        }
        moves++;
    }
    printf("Your score is %d moves.\n",moves);
    return 0;
}